src/compile/data/formatparse.ts

```diff
@@ -74,6 +79,11 @@ export class ParseNode extends DataFlowNode {
             return;
           }
           parse[fieldDef.field] = 'number';
+        } else if (countAccessPath(fieldDef.field) > 1) {
```

Comment:
countAccessPath is a pretty confusing name.  

`accessPathDepth`?

Reply:
Also why is this case checked after `isTimeFieldDef` / `isNumberFieldDef`? 

What if a number field def has nested field?

Reply:
Then we flatten the field already. Note that every expression flattens.